simple_certmanager/csr_generation.py

```diff
@@ -0,0 +1,70 @@
+from cryptography import x509
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import hashes, serialization
+from cryptography.hazmat.primitives.asymmetric import rsa
+from cryptography.x509 import NameOID
+
+# These methods make use of the SigningRequest model to generate a private key and CSR
+# It is not imported here to avoid circular imports
```

Comment:
This is exactly the part I want to avoid by putting the code in a separate module. This should be pure python and be decoupled entire from Django, since nothing about privatey key/CSR generation is tied to the Django framework.

So, instead of operating on a model instance, you should be taking in the raw primitives, and in the model layer make the changes to call them. The code suggestions below demonstrate a cleaner way to manage this separation of concerns.